spark/src/main/java/org/apache/iceberg/spark/actions/BaseSparkAction.java

```diff
@@ -167,19 +133,21 @@ protected Table newStaticTable(TableMetadata metadata, FileIO io) {
   }
 
   protected Dataset<Row> buildManifestListDF(Table table) {
-    List<String> manifestLists = getManifestListPaths(table.snapshots());
+    List<String> manifestLists = ReachableFileUtil.manifestListLocations(table);
     return spark.createDataset(manifestLists, Encoders.STRING()).toDF("file_path");
   }
 
-  protected Dataset<Row> buildOtherMetadataFileDF(TableOperations ops) {
-    List<String> otherMetadataFiles = getOtherMetadataFilePaths(ops);
-    return spark.createDataset(otherMetadataFiles, Encoders.STRING()).toDF("file_path");
+  protected Dataset<Row> buildOtherMetadataFileDF(Table table) {
+    Set<String> otherMetadataFiles = ReachableFileUtil.metadataFileLocations(table, false);
```

Comment:
nit: I think it is a little confusing to call metadata file locations as otherMetadataFiles and then add version hint to it. Plus, the set returned from the utility may be immutable in the future. 

What about this?

```
List<String> otherMetadataFiles = Lists.newArrayList();
otherMetadataFiles.addAll(ReachableFileUtil.metadataFileLocations(table, false));
otherMetadataFiles.add(ReachableFileUtil.versionHintLocation(table));
return spark.createDataset(otherMetadataFiles, Encoders.STRING()).toDF("file_path");
```